inlong-manager/manager-client/src/main/java/org/apache/inlong/manager/client/api/impl/InlongGroupImpl.java

```diff
@@ -294,17 +294,25 @@ private InlongGroupContext generateSnapshot() {
     private InlongGroupContext generateSnapshot(String credentials) {
         InlongGroupContext groupContext = generateSnapshot();
         InlongGroupInfo groupInfo = groupContext.getGroupInfo();
-        if (groupInfo.getExtList().stream().anyMatch(info -> InlongConstants.SORT_JOB_ID.equals(info.getKeyName())
-                && StringUtils.isNotEmpty(info.getKeyValue()))) {
-            ListSortStatusRequest request = new ListSortStatusRequest();
+        if (groupInfo.getExtList().stream().anyMatch(ext -> InlongConstants.SORT_JOB_ID.equals(ext.getKeyName())
+                && StringUtils.isNotEmpty(ext.getKeyValue()))) {
+            SortStatusRequest request = new SortStatusRequest();
             request.setInlongGroupIds(Collections.singletonList(groupInfo.getInlongGroupId()));
             request.setCredentials(credentials);
-            ListSortStatusResponse sortStatusInfo = groupClient.listSortStatus(request);
-            if (MapUtils.isNotEmpty(sortStatusInfo.getStatusMap())) {
-                groupContext.updateSortStatus(sortStatusInfo.getStatusMap().getOrDefault(
-                        groupInfo.getInlongGroupId(), SortStatus.UNKNOWN));
+            List<SortStatusInfo> statusInfos = groupClient.listSortStatus(request);
+
+            SortStatus sortStatus = SortStatus.UNKNOWN;
+            if (CollectionUtils.isNotEmpty(statusInfos)) {
+                Optional<SortStatusInfo> optional = statusInfos.stream()
+                        .filter(status -> groupInfo.getInlongGroupId().equals(status.getInlongGroupId()))
+                        .findFirst();
+                if (optional.isPresent()) {
+                    sortStatus = optional.get().getSortStatus();
+                }
             }
+            groupContext.updateSortStatus(sortStatus);
         }
+
```

Comment:
Suggest updating status to NOT_EXISTS rather than the default UNKNOWN status at the else branch  of the If statement.

Reply:
done